bindings/python3/econf.py

```diff
@@ -0,0 +1,996 @@
+"""
+Econf provides functionality for interacting with Key-Value config files, like getting and setting values for read config files.
+
+For more information please have a look at the API
+"""
+import ctypes.util
+from enum import Enum
+from dataclasses import dataclass
+from typing import *
+from ctypes import *
+
+LIBNAME = ctypes.util.find_library("econf")
+LIBECONF = CDLL(LIBNAME)
+
+
+@dataclass
+class EconfFile:
+    """
+    Class which points to the Key Value storage object
+    """
+
+    _ptr: c_void_p
+
+    def __del__(self):
+        free_file(self)
+
+
+class Econf_err(Enum):
+    ECONF_SUCCESS = 0
+    ECONF_ERROR = 1
+    ECONF_NOMEM = 2
+    ECONF_NOFILE = 3
+    ECONF_NOGROUP = 4
+    ECONF_NOKEY = 5
+    ECONF_EMPTYKEY = 6
+    ECONF_WRITEERROR = 7
+    ECONF_PARSE_ERROR = 8
+    ECONF_MISSING_BRACKET = 9
+    ECONF_MISSING_DELIMITER = 10
+    ECONF_EMPTY_SECTION_NAME = 11
+    ECONF_TEXT_AFTER_SECTION = 12
+    ECONF_FILE_LIST_IS_NULL = 13
+    ECONF_WRONG_BOOLEAN_VALUE = 14
+    ECONF_KEY_HAS_NULL_VALUE = 15
+    ECONF_WRONG_OWNER = 16
+    ECONF_WRONG_GROUP = 17
+    ECONF_WRONG_FILE_PERMISSION = 18
+    ECONF_WRONG_DIR_PERMISSION = 19
+    ECONF_ERROR_FILE_IS_SYM_LINK = 20
+    ECONF_PARSING_CALLBACK_FAILED = 21
+
+
+def _exceptions(err: int, val: str):
+    if Econf_err(err) == Econf_err.ECONF_ERROR:
+        raise Exception(val)
```

Comment:
Ok. I am starting to dislike this approach.

(not tested)

```python
ECONF_EXCEPTION = {
    EconfError.ERROR: Exception,
    EconfError.NOMEN: MemoryError,
    EconfError.NOFILE: FileNotFound,
    ...
}

raise ECONF_EXCEPTION[NOFILE](f"File {path} not found")
```

Reply:
I think I tested this before and python doesn't let you raise exceptions like this by calling the value of a dict. But let me confirm this again

Reply:
check the code, it is returning the exception class, no the exception instance